doc/developer/design/00000000_template.md

```diff
@@ -1,42 +1,93 @@
--->
+One paragraph to explain the feature.
 
-## Goals
+# Motivation
+[motivation]: #motivation
 
-<!--
-// Enumerate the concrete goals that are in scope for the project.
--->
+Why are we doing this? What problems does it solve? What use cases does it enable? What is the desired outcome?
 
-## Non-Goals
+# Explanation
+[explanation]: #explanation
 
-<!--
-// Enumerate potential goals that are explicitly out of scope for the project
-// ie. what could we do or what do we want to do in the future - but are not doing now
--->
+Explain the design as if it were part of Materialize and you were teaching the team about it.
+This can mean:
 
-## Description
+- Introduce new named concepts.
+- Explain the feature using examples that demonstrate product-level changes.
+- Explain how it builds on the current architecture.
+- Explain how engineers and users should think about this change, and how it influences how everyone uses the product.
+- If needed, talk though errors, backwards-compatibility, or migration strategies.
+- Discuss how this affects maintainability, or whether it introduces concepts that might be hard to change in the future.
 
-<!--
-// Describe the approach in detail. If there is no clear frontrunner, feel free to list all approaches in alternatives.
-// If applicable, be sure to call out any new testing/validation that will be required
--->
+# Reference explanation
+[reference-explanation]: #reference-explanation
 
-## Alternatives
+Focus on the implementation of the feature.
+This is the technical part of the design.
 
-<!--
-// Similar to the Description section. List of alternative approaches considered, pros/cons or why they were not chosen
--->
+- Is it reasonably clear how the feature is implemented?
+- What dependencies does the feature have and introduce?
+- Focus on corner cases.
+- How can we test the feature and protect against regressions?
 
-## Open questions
+# Rollout
+[rollout]: #rollout
 
-<!--
-// Anything currently unanswered that needs specific focus. This section may be expanded during the doc meeting as
-// other unknowns are pointed out.
-// These questions may be technical, product, or anything in-between.
--->
+Describe what steps are necessary to enable this feature for users.
+
+## Testing
+[testing]: #testing
+
+Describe how you will test and roll out the implementation.
+When the deliverable is a refactoring, the existing tests may be sufficient.
+When the deliverable is a new feature, new tests are imperative.
+
+Basic guidelines:
+
+* Nearly every feature requires either Rust unit tests, sqllogictest tests, or testdrive tests.
+* Features that interact with Kubernetes additionally need a cloudtest test.
+* Features that interact with external systems additionally should be tested manually in a staging environment.
+* Features or changes to performance-critical parts of the system should be load tested.
+
+## Lifecycle
```

Comment:
Maybe fits under the `Testing` section, but I'd add some reference around ongoing monitoring and observability